components/sensor/collector.go

```diff
@@ -50,17 +40,17 @@ func newSensorCollector(resource interface{}, params data.CollectorParams) (data
 			}
 			return nil, data.FailedToReadErr(params.ComponentName, readings.String(), err)
 		}
+		readings := make(map[string]*structpb.Value)
 		for name, value := range values {
-			if len(arg) != 0 {
-				// If specific sensor reading names were passed in the robot config, report only those
-				// Otherwise, report all sensor values
-				if _, ok := arg[name]; !ok {
-					continue
-				}
+			val, err := structpb.NewValue(value)
```

Comment:
Spoke about this with @agreenb. We decide to bring sensor more in line with the API, we will no longer be filtering on sensor data, the collector will always get all values from the sensor

Reply:
Note that before merging this PR, will post in #eng-breaking-changes and #solutions-engineering.